src/pages/how-does-react-tell-a-class-from-a-function/index.nl.md

```diff
@@ -0,0 +1,553 @@
+
+```jsx
+const Greeting = () => <p>Hello</p>;
+new Greeting(); // 🔴 Greeting is geen constructor
+```
+
+Dit is express gedaan en heeft te maken met het ontwerp van arrow functies. Een van de grootste voordelen van een arrow functie is dat ze niet hun eigen `this` value hebben - in plaats daarvan komt `this` van de dichtstbijzijnde normale functie:
+
+```jsx{2,6,7}
+class Friends extends React.Component {
+  render() {
+    const friends = this.props.friends;
+    return friends.map(friend =>
+      <Friend
+        // `this` komt uit de ‘render’ method.
+        size={this.props.size}
+        name={friend.name}
+        key={friend.id}
+      />
+    );
+  }
+}
+```
+
+Oke, dus **arrow functies hebben geen beschikking over hun eigen `this`.** Wacht... dat zou betekenen dat ze totaal onbruikbaar zijn als constructors!
+
+```jsx
+const Person = (name) => {
+  // 🔴 Dit zou niet logisch zijn
+  this.name = name;
+}
+```
+
+Daarom **maakt JavaScript het onmogelijk om een arrow function aan te roepen met `new`.** Als je dit wel zou doen is de kans dat je een fout maakt toch al aanwezig. Dat kan je dan maar beter zo snel mogelijk weten. Het is een beetje hetzelfde als hoe JavaScript het je niet toelaat om een class aan te roepen zonder `new`.
+
+Heel leuk, maar het maakt ons plan wel iets moeilijker. React kan niet zomaar `new` aanroepen op ieder type functie omdat het mis zou gaan bij arrow functies. We zouden kunnen proberen om arrow functies te filteren door te kijken of ze geen `prototype` hebben en hierdoor besluiten of we wel of niet `new` kunnen gebruiken:
+
+```jsx
+(() => {}).prototype // undefined
+(function() {}).prototype // {constructor: f}
+```
+
+Maar dit [zou niet werken](https://github.com/facebook/react/issues/4599#issuecomment-136562930) voor functies die al zijn compiled door Babel. Misschien niet echt een big deal, maar er is ook nog een andere reden die ervoor zorgt dat deze aanpak niet een goed idee is.
+
+---
+
+Een andere reden waarom we niet altijd `new` kunnen gebruiken is omdat het React ervan zou weerhouden om components te ondersteunen die strings of andere primitieve types teruggeven.
+
+```jsx
+function Greeting() {
+  return 'Hello';
+}
+
+Greeting(); // ✅ 'Hello'
+new Greeting(); // 😳 Greeting {}
+```
+
+Dit heeft wederom te maken met de gekkigheden van het ontwerp van de [`new` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/new). Zoals we eerder zagen vertelt `new` de JavaScript engine om een object te maken, deze om te zetten in `this` binnen de functie en dit object later terug te geven als resultaat van `new`.
+
+Echter, JavaScript staat het functies die zijn aangeroepen met `new` ook toe om de return value van `new` te overschrijven door een ander object terug te geven. Waarschijnlijk omdat dit handig zou zijn voor patterns zoals pooling waarbij we instances willen hergebruiken:
```

Comment:
```suggestion
Echter, JavaScript staat het functies die zijn aangeroepen met `new` ook toe om de return value van `new` te overschrijven door een ander object terug te geven. Waarschijnlijk omdat dit handig zou zijn voor patterns zoals pooling waarbij we instanties willen hergebruiken:
```

Not sure if we should translate it or keep it in English. We do have a good Dutch word for "instances", but it might be less clear to the reader. The same goes for "gecompiled" and "gecompileerd".